bitcoin/src/crypto/key.rs

```diff
@@ -71,6 +72,13 @@ impl PublicKey {
         }
     }
 
+    /// Returns the script code used to spend a P2WPKH input.
+    pub fn p2wpkh_script_code(&self) -> Result<ScriptBuf, UncompressedPubkeyError> {
+        let wpkh = self.wpubkey_hash()?;
+        let spk = ScriptBuf::new_p2wpkh(&wpkh);
+        Ok(spk.p2wpkh_script_code().expect("spk is a p2wpkh"))
```

Comment:
I'd like to avoid intermediate allocation and panic in the future but it's OK now.

Reply:
Yes, both could be eliminated by manually constructing the scriptCode rather than going through the `p2wpkh_script_code` method. And I think the function for `PublicKey` could try the conversion to a `CompressedPublicKey` and then call `CompressedPublicKey::p2wpkh_script_code`, which would reduce code duplication.

I'll ACK this but will wait for Tobin to see if he wants to address this in this PR or leave it for a later optimization.